provider/pkg/schema/gen_tags.go

```diff
@@ -44,14 +45,41 @@ func (ctx *context) GetTagsStyle(typeSpec *pschema.TypeSpec, originalSpec *jssch
 	if typeSpec.Ref == "pulumi.json#/Any" {
 		return TagsStyleUntyped
 	}
-	switch typeSpec.Type {
-	case "object":
-		if typeSpec.AdditionalProperties != nil && typeSpec.AdditionalProperties.Type == "string" {
-			return TagsStyleStringMap
-		} else {
-			return TagsStyleUnknown
+
+	if typeSpec.AdditionalProperties != nil && typeSpec.AdditionalProperties.Type == "string" {
+		return TagsStyleStringMap
+	}
+
+	if ctx.TagStyleIsKeyValueArray(typeSpec, originalSpec) {
+		return TagsStyleKeyValueArray
+	}
+
+	return TagsStyleUnknown
+}
+
+func (ctx *context) TagStyleIsKeyValueArray(typeSpec *pschema.TypeSpec, originalSpec *jsschema.Schema) bool {
+	if typeSpec == nil || typeSpec.Items == nil || typeSpec.Items.Ref == "" {
+		return false
+	}
+
+	typeToken, hasPrefix := strings.CutPrefix(typeSpec.Items.Ref, "#/types/")
+	if !hasPrefix {
+		return false
+	}
+
+	if refType, ok := ctx.pkg.Types[typeToken]; ok {
+		keyProp, keyPropExists := refType.Properties["key"]
+		valueProp, valuePropExists := refType.Properties["value"]
+		// Check if the type has exactly two properties, "key" and "value", both of type "string"
+		for k := range refType.Properties {
```

Comment:
[nit] This loop could just be `keyPropExists && valuePropExists && len(refType.Properties) == 2`, right?

Reply:
Ah yup - much nicer 👍